libpod/container_internal_linux.go

```diff
@@ -1717,11 +1717,27 @@ func (c *Container) generateCurrentUserPasswdEntry() (string, int, int, error) {
 	// If the user's actual home directory exists, or was mounted in - use
 	// that.
 	homeDir := c.WorkingDir()
-	if MountExists(c.config.Spec.Mounts, u.HomeDir) {
-		homeDir = u.HomeDir
+	hDir := u.HomeDir
+	for hDir != "/" {
```

Comment:
u.HomeDir is a string no?
```suggestion
	if hDir != "/" {
```

Reply:
This code is attempting to walk up the directories to check if their is a mountpoint from the host that covers this directory.

For example if my homedir was /home/engineering/dwalsh.
I want to check if 
 /home/engineering/dwalsh
 /home/engineering
 /home
Is mounted into the container, if yes then I can use /home/engineering/dwalsh as the homedir, rather then CWD.

Reply:
No, we want to look through the directories.

Reply:
I don't know if this is really wanted - when I originally wrote this, I wanted it to be specific to ensure that we didn't leak information about the user on the host into the container unless they were explicitly mounting their home